testsuite/test_cim_operations2.py

```diff
@@ -0,0 +1,142 @@
+
+        # correctly encoded but ill-formed UTF-8
+        if self.VERBOSE:
+            print("From here on, the only expected exception is ParseError "
+                  "for ill-formed UTF-8 Byte sequences...")
+        # combo of U+D800,U+DD22:
+        self._run_single(b'<V>a\xED\xA0\x80\xED\xB4\xA2b</V>', False)
+        # combo of U+D800,U+DD22 and combo of U+D800,U+DD23:
+        # pylint: disable=line-too-long
+        self._run_single(
+            b'<V>a\xED\xA0\x80\xED\xB4\xA2b\xED\xA0\x80\xED\xB4\xA3</V>',
+            False)
+
+        # incorrectly encoded UTF-8
+        if self.VERBOSE:
+            print("From here on, the only expected exception is ParseError "
+                  "for invalid UTF-8 Byte sequences...")
+        # incorrect 1-byte sequence:
+        self._run_single(b'<V>a\x80b</V>', False)
+        # 2-byte sequence with missing second byte:
+        self._run_single(b'<V>a\xC0', False)
+        # 2-byte sequence with incorrect 2nd byte
+        self._run_single(b'<V>a\xC0b</V>', False)
+        # 4-byte sequence with incorrect 3rd byte:
+        self._run_single(b'<V>a\xF1\x80abc</V>', False)
+        # 4-byte sequence with incorrect 3rd byte that is an incorr. new start:
+        self._run_single(b'<V>a\xF1\x80\xFFbc</V>', False)
+        # 4-byte sequence with incorrect 3rd byte that is an correct new start:
+        self._run_single(b'<V>a\xF1\x80\xC2\x81c</V>', False)
+
+
+class TestGetRsltParams(object):
+    """Test WBEMConnection._get_rslt_params method."""
+
+    @pytest.mark.parametrize(
+        'irval, ec, eos, eosr', [[None, u'contextblah', u'False', False],
+                                 [None, "", u'True', True],
+                                 [None, u'contextblah', u'True', True]]
+    )
+    @staticmethod
+    def test_pass(irval, ec, eos, eosr):
+        """Test good combinations of EndOfSequence and EnumerationContext"""
+        result = [
+            (u'IRETURNVALUE', {}, irval),
+            (u'EnumerationContext', None, ec),
+            (u'EndOfSequence', None, eos)
+        ]
+        # pylint: disable=protected-access
+        result = WBEMConnection._get_rslt_params(result, 'namespace')
+        # the _get_rslt_params method sets context to None if eos True
+        ecs = None if eosr is True else (ec, 'namespace')
+        assert result == (None, eosr, ecs)
+
+    @pytest.mark.parametrize(
+        'irval, ec, eos', [[None, None, None],
+                           [None, None, u'False']]
+    )
+    @staticmethod
+    def test_fail(irval, ec, eos):
+        """Test EndOfSequence and EnumerationContext variations that should
+           cause exception."""
+        result = [
+            (u'IRETURNVALUE', {}, irval),
+            (u'EnumerationContext', None, ec),
+            (u'EndOfSequence', None, eos)
+        ]
+        with pytest.raises(CIMError):
+            # pylint: disable=protected-access
+            result = WBEMConnection._get_rslt_params(result, 'namespace')
```

Comment:
I suggest to also assert the status code of the `CIMError` that is raised:
```
        with pytest.raises(CIMError) as exc_info:
            result = . . .
        exc = exc_info.value
        assert exc.status_code_name == 'CIM_ERR_. . .'
```

Reply:
Done.